gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -230,47 +191,114 @@ var gpii = fluid.registerNamespace("gpii");
         return fluid.invokeGradedFunction(resolvedName, action);
     };
 
+    /** Compensate for the effect of simpleminded merging when applied to a snapshot where an "DELETE" is merged on top of an "ADD"
+     */
+    gpii.lifecycleManager.cleanDeletes = function (value) {
+        if (value.type === "DELETE") {
+            delete value.value;
+        }
+        return value;
+    };
+
+    /** Applies snapshotted settings from a single settingsHandler block attached to a single solution into the "originalSettings"
+     * model snapshot area in the LifecycleManager's session. Tightly bound to executeSettingsAction, executes one-to-one with it with
+     * almost identical argument list.
+     */
+
+    gpii.lifecycleManager.processSnapshot = function (that, snapshot, solutionId, solutionRecord, session, settingsHandlerBlockName, rootAction) {
```

Comment:
Could we use a more descriptive name here rather than `processSnapshot`? Maybe something like `recordSnapshotInSession` or `applySnapshotToSessionSettings`.